src/discord-cluster-manager/cogs/leaderboard_cog.py

```diff
@@ -251,76 +273,132 @@ async def leaderboard_create(
             try:
                 date_value = datetime.strptime(deadline, "%Y-%m-%d %H:%M")
             except ValueError:
-                # If that fails, try parsing just the date (will set time to 00:00)
-                date_value = datetime.strptime(deadline, "%Y-%m-%d")
+                try:
+                    date_value = datetime.strptime(deadline, "%Y-%m-%d")
+                except ValueError:
+                    raise ValueError(
+                        "Invalid date format. Please use YYYY-MM-DD or YYYY-MM-DD HH:MM"
+                    )
 
+            # Ask the user to select GPUs
+            view = GPUSelectionView([gpu.name for gpu in GitHubGPU])
+
+            await interaction.response.send_message(
+                "Please select GPUs for this leaderboard:",
+                view=view,
+                ephemeral=True,
+            )
+
+            # Wait until the user makes a selection
+            await view.wait()
+
+        except ValueError as ve:
+            # Handle invalid date errors
+            await interaction.response.send_message(
+                str(ve),
+                ephemeral=True,
+            )
+
+        # Kind of messy, but separate date try/catch
+        try:
             # Read the template file
             template_content = await reference_code.read()
 
             with self.bot.leaderboard_db as db:
-                print(
-                    leaderboard_name,
-                    type(date_value),
-                    type(template_content.decode("utf-8")),
-                )
-                db.create_leaderboard({
+                err = db.create_leaderboard({
                     "name": leaderboard_name,
                     "deadline": date_value,
                     "reference_code": template_content.decode("utf-8"),
+                    "gpu_types": view.selected_gpus,
                 })
 
-            await interaction.response.send_message(
+                if err:
+                    print(err)
```

Comment:
```suggestion
```
...or log the error.

Reply:
Removed these, this should not have been in the commit. I was debugging at the time, thanks for catching!